include/datadog/telemetry/telemetry.h

```diff
@@ -78,4 +73,89 @@ void report_error_log(std::string message);
 /// @param stacktrace Stacktrace leading to the error.
 void report_error_log(std::string message, std::string stacktrace);
 
+/// The `counter` namespace provides functions to track values.
+/// Counters can be useful for tracking the total number of an event occuring in
+/// one time interval. For example, the amount of requests, errors or jobs
+/// processed every 10 seconds.
+namespace counter {
+
+/// Increments the specified counter by 1.
+///
+/// @param `counter` the counter to increment.
+void increment(const Counter& counter);
+
+/// Increments the specified counter by 1.
+///
+/// @param `counter` the counter to increment.
+/// @param `tags` the distribution tags.
+void increment(const Counter& counter, const std::vector<std::string>& tags);
```

Comment:
minor: should we consider spans of string views ?

Reply:
spans, like `std::spans`? If so, unfortunately we are stuck with C++17 at the moment.

Reply:
Ah yes, we used to rely on [tcb spans](https://github.com/DataDog/ddprof/blob/78b6920991c720b7c9f2b21af07f71ab3ae8ada0/third_party/tcb/span.hpp) when we did not have c++ 20
Vectors seem fine in the mean time